apps/sample-app/src/components/Calendar.tsx

```diff
@@ -1,21 +1,57 @@
 import './styles.css';
 
 import { Button } from '@fluentui/react-components';
-import { calendar } from '@microsoft/teams-js';
+import { app, calendar } from '@microsoft/teams-js';
 import React from 'react';
 
-export const Calendar: React.FunctionComponent = () => {
-  const handleCalendar = async (): Promise<void> => {
-    // create calendar paramaters
+import { MessageListItem } from './Emails';
+
+const teamsDeepLinkHost = 'teams.microsoft.com';
+const teamsDeepLinkProtocol = 'https';
+const teamsDeepLinkAttendeesUrlParameterName = 'attendees';
+const teamsDeepLinkUrlPathForCalendar = '/l/meeting/new';
+
+export const handleNewMail = async (): Promise<void> => {
+  if (!calendar.isSupported()) {
+    const context = await app.getContext();
+    if (context?.app?.host?.name === 'Teams') {
```

Comment:
Hmm I think we want to avoid checking for a specific host? Was this the workaround you mentioned today?

Reply:
oh right! I just ran the code without any host check. It works perfectly okay, my initial reasoning for the check was 'what if its running on office and office doesn't support composing  a new meeting' but it turns out it redirects you to Teams anyway. Thanks!!